### PR TITLE
changes for timeInterval

### PR DESCRIPTION
**Pull request by hanbollar**
Fixes [#6164](https://github.com/AnalyticalGraphicsInc/cesium/issues/6164)

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Signed CLA is on file.

@hanbollar, thanks for the pull request! Maintainers, we have a signed CLA from @hanbollar, so you can review this at any time.

---

__I am a bot who helps you make Cesium awesome! [Contributions to my configuration are welcome.](https://github.com/AnalyticalGraphicsInc/cesium/tree/master/.concierge)__

:earth_africa: :earth_americas: :earth_asia:

**Comment by hanbollar** [comment]
@hpinkos I updated the documentation and added in the throw statement. lmk if this is what you were looking for / if there's anything else you think I should do.

**Comment by hanbollar** [comment]
@mramato updated to match that

**Comment by ggetz** [comment]
Thanks @hanbollar, can you add a quick test as well?

File: Specs/Core/TimeIntervalSpec.js [hunk 1]
```diff
@@ -100,6 +100,12 @@ defineSuite([
         expect(interval.data).toEqual(data);
     });
 
+    it('fromIso8601 throws error when given Invalid ISO 8601 date.', function() {
+        expect(function() {
+            expect(TimeInterval.fromIso8601({ iso8601 : '2020-08-29T00:00:00+00:00' }));
```
Comment (ggetz) [inline Specs/Core/TimeIntervalSpec.js]: You don't need to wrap this in the second `expect`:

```
expect(function() {
    TimeInterval.fromIso8601({ iso8601 : '2020-08-29T00:00:00+00:00' });
});
```

**Comment by hanbollar** [comment]
@ggetz removed second expect

**Comment by ggetz** [comment]
Thanks @hanbollar !

File: Source/Core/TimeInterval.js [hunk 2]
```diff
@@ -143,6 +147,9 @@ define([
         //>>includeEnd('debug');
 
         var dates = options.iso8601.split('/');
+        if (dates.length !== 2) {
+            throw new DeveloperError('Invalid ISO 8601 date.');
```
Comment (hpinkos) [inline Source/Core/TimeInterval.js]: `options.iso8601 is an invalid ISO 8601 interval`

**Pull request merged**